pkg/project/project.go

```diff
@@ -219,29 +235,41 @@ func generateProjectName() string {
 	return strings.Join(str, " ")
 }
 
-// FindFileOrDirectory searches for a file or directory with name `filename`.
-// Search starts in `startDir` and will traverse through all parent directories until the file is found,
-// root directory is reached or `maxRecursiveIteration` is exceeded.
-func FindFileOrDirectory(startDir, fileDir, filename string) (string, bool) {
+// FindFileOrDirectory searches for a file or directory named `filename`.
+// Search starts in `directory` and will traverse through all parent directories.
+// `directory` may also be a file, and in that case will start from the file's directory.
+func FindFileOrDirectory(directory, filename string) (string, bool) {
 	i := 0
 	for i < maxRecursiveIteration {
-		if fileExists(filepath.Join(startDir, fileDir, filename)) {
-			return filepath.Join(startDir, fileDir, filename), true
+		if isRootPath(directory) {
+			return "", false
 		}
 
-		startDir = filepath.Clean(filepath.Join(startDir, ".."))
-		if startDir == "." || startDir == "/" || driveLetterRegex.MatchString(startDir) {
-			return "", false
+		// remove any trailing file from directory
+		directory = filepath.Dir(directory + string(filepath.Separator))
```

Comment:
Previously the first iteration would skip the current directory, for ex:

```go
 FindFileOrDirectory("/path/to/file", ".wakatime-project")
```

Would only start looking for `.wakatime-project` at `/path/.wakatime-project` missing any file at `/path/to/.wakatime-project`. This might have caused the bug seen in #575.

Reply:
Not sure it would happen because of current implementation. Did you get this behavior on tests?

Reply:
Checked and you're right, it makes no difference. I remember I was going to change `filepath.Join(startDir, "..")` to `filepath.Dir(startDir)` but using `Join` with `".."` avoids the whole problem.